lib/src/modules/task.cc

```diff
@@ -426,9 +438,93 @@ static fs::path getCachedTaskFile(const fs::path& task_cache_dir,
     }
 }
 
+// build a spool dir for the supporting library files requested by a multifile task
+static fs::path createInstallDir(const fs::path& spool_dir, std::set<std::string> download_set) {
```

Comment:
It appears that the executable for a single file task is being run from it's location in the cache. I noticed that spool dir is configurable so I am assuming we want to use that. That will likely be a new parameter in the `Task` constructor. I want to verify that is the correct use of spool dir before altering the constructor.

Reply:
I think the spool dir will already have been created. The directory for `_installdir` should be a subdirectory under whatever spooldir will be used for the execution of this action.

Reply:
I think creating the install dir from downloaded files could happen outside of the `Task` class.

I think in `callAction`, you can get the results dir (which is what I mean by spool_dir) from the request object.

Reply:
One twist is that we only create a resultsDir for a`RequestType::Blocking`. For non-blocking requests, we probably just want to create a tempdir for `_installdir`.